### ISSUE TITLE
Menu.Target Has Accessibility Issue

### ISSUE THREAD

**Issue by bcaylor10** [mantinedev/mantine#6349]
### Dependencies check up

- [X] I have verified that I use latest version of all @mantine/* packages

### What version of @mantine/* packages do you have in package.json?

7.10.1

### What package has an issue?

@mantine/core

### What framework do you use?

create-react-app (CRA)

### In which browsers you can reproduce the issue?

All

### Describe the bug

The Menu.Target component is throwing an accessibility error using the Axe accessibility tool extension. The error I get is:

"Ensures an element's role supports its ARIA attributes"

```
<div class="styles_menuTarget__6I6Wy" aria-haspopup="menu" aria-expanded="false" aria-controls="mantine-dcrjgoud5-dropdown" id="mantine-dcrjgoud5-target">
```

"To solve this problem, you need to fix the following:

ARIA attribute is not allowed: aria-expanded="false"

### If possible, include a link to a codesandbox with a minimal reproduction

Not a sandbox link: https://fitnessconnection.com/join-new#/new

### Possible fix

According to the documentation on the Mantine site, the [accessibility section](https://mantine.dev/core/menu/#accessibility) links to [this page](https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/examples/menu-button-links/) which states that the element should be a button. So, I believe replacing the div or adding a role of button will fix this issue.

### Self-service

- [ ] I would be willing to implement a fix for this issue

**Comment by bcaylor10** [comment]
I am having the same issue with the Popover.Target component

**Comment by rtivital** [comment]
The issue is on your side. Menu.Target does not render any elements, it supplies props to the element that it receives in `children`. Change the element to be `<button />` on your side.
